src/full/Agda/TypeChecking/Unquote.hs

```diff
@@ -734,3 +735,14 @@ evalTCM v = do
       let i = mkDefInfo (nameConcrete $ qnameName x) noFixity' PublicAccess ConcreteDef noRange
       checkFunDef NotDelayed i x cs
       primUnitUnit
+
+    tcRunSpeculative :: Term -> UnquoteM Term
+    tcRunSpeculative mu = do
+      oldState <- getTC
+      u <- reduce =<< evalTCM mu
+      case u of
+        Con _ _ [Apply (Arg { unArg = x }), Apply (Arg { unArg = b })] -> do
+          unlessM (unquote b) $ putTC oldState
+          return x
+        _ -> liftTCM $ typeError . GenericDocError =<<
+          "Should be a pair: " <+> prettyTCM u
```

Comment:
Given the type of the `BUILTIN`, shouldn't this be an `__IMPOSSIBLE__`?

Reply:
If the user is silly, it could be a postulate or something..